plugins/com.google.cloud.tools.eclipse.jdt.launching/src/com/google/cloud/tools/eclipse/jdi/internal/connect/SocketListeningMultiConnectorImpl.java

```diff
@@ -0,0 +1,208 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Ivan Popov - Bug 184211: JDI connectors throw NullPointerException if used separately
+ *     			from Eclipse
+ *     Google Inc - add support for accepting multiple connections
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.jdi.internal.connect;
+
+import com.sun.jdi.VirtualMachine;
+import com.sun.jdi.connect.Connector;
+import com.sun.jdi.connect.IllegalConnectorArgumentsException;
+import com.sun.jdi.connect.ListeningConnector;
+import com.sun.jdi.connect.Transport;
+
+import org.eclipse.jdi.internal.VirtualMachineManagerImpl;
+import org.eclipse.jdi.internal.connect.ConnectMessages;
+import org.eclipse.jdi.internal.connect.ConnectorImpl;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A fork of
+ * {@link org.eclipse.jdi.internal.connect.SocketListeningConnectorImpl} that
+ * uses our modified SocketTransportImpl.
+ */
+public class SocketListeningMultiConnectorImpl extends ConnectorImpl implements ListeningConnector {
+	/** Port to which is attached. */
+	private int fPort;
+	/** Timeout before accept returns. */
+	private int fTimeout;
+
+	/**
+	 * Creates new SocketAttachingConnectorImpl.
+	 */
+	public SocketListeningMultiConnectorImpl(VirtualMachineManagerImpl virtualMachineManager) {
+		super(virtualMachineManager);
+
+		// Create communication protocol specific transport.
+		SocketTransportImpl transport = new SocketTransportImpl();
+		internalSetTransport(transport);
+	}
+
+	private void internalSetTransport(Transport transport) {
+		try {
+			Method setter = ConnectorImpl.class.getDeclaredMethod("setTransport", Transport.class);
+			setter.setAccessible(true);
+			setter.invoke(this, transport);
+		} catch (Exception ex) {
+			// probably should log this
+			ex.printStackTrace();
+		}
+	}
+
+	/**
+	 * @return Returns the default arguments.
+	 */
+	@SuppressWarnings("restriction")
+	@Override
+	public Map<String, Connector.Argument> defaultArguments() {
+		HashMap<String, Connector.Argument> arguments = new HashMap<String, Connector.Argument>(1);
+
+		// Port
+		IntegerArgument intArg = new _IntegerArgumentImpl("port", //$NON-NLS-1$
+				ConnectMessages.SocketListeningConnectorImpl_Port_number_at_which_to_listen_for_VM_connections_1,
+				ConnectMessages.SocketListeningConnectorImpl_Port_2, true, SocketTransportImpl.MIN_PORTNR,
+				SocketTransportImpl.MAX_PORTNR);
+		arguments.put(intArg.name(), intArg);
+
+		// Timeout
+		intArg = new _IntegerArgumentImpl("timeout", //$NON-NLS-1$
+				ConnectMessages.SocketListeningConnectorImpl_Timeout_before_accept_returns_3,
+				ConnectMessages.SocketListeningConnectorImpl_Timeout_4, false, 0, Integer.MAX_VALUE);
+		arguments.put(intArg.name(), intArg);
+
+		// FIXME: this doesn't feel like the right place, but
```

Comment:
Good question.  In the `SocketListeningMultiConnector` — it's for the `SocketListeningMultiConnectorProcess`.  An alternative is to open up the `*ArgumentImpl`s.